ddtrace/tracer/span.go

```diff
@@ -289,10 +311,14 @@ func (s *span) setMetric(key string, v float64) {
 	}
 	delete(s.Meta, key)
 	switch key {
+	case ext.ManualKeep:
+		if v == float64(samplerAppSec) {
+			s.setSamplingPriorityLocked(ext.PriorityUserKeep, samplerAppSec, math.NaN())
+		}
```

Comment:
> Doesn't this case cause ext.ManualKeep to not be set unless by the samplerAppSec?

No, `span.SetTag(ext.ManualKeep, true)` still works as before.

> Do we want ext.ManualKeep to work for other samplers?

I think we shouldn't support/allow customer specifying the sampler freely. This was a small hack for AppSec, because AppSec library cannot access private properties/methods of the tracer.

Reply:
> No, span.SetTag(ext.ManualKeep, true) still works as before.

I know that still works. What I mean is that setting manual keep with any other sampler value will silently fail, and unless I read the `setMetric` code, I have no way to know that. This will be surprising behavior if I see appsec setting the sampler with this mechanism.

If we want to do this hack, we should expose `samplerAppSec` as some `internal` constant so that appsec can use it. I guess it would have to be in `github.com/DataDog/dd-trace-go.v1/internal`. We should also document the mechanism (not in the public docs) so we know what works (only `samplerAppSec`)